analyzers/src/SonarAnalyzer.CSharp/Rules/DisposeFromDispose.cs

```diff
@@ -87,25 +87,25 @@ protected override void Initialize(SonarAnalysisContext context)
         /// </remarks>
         private static bool IsDisposeMethodCalled(InvocationExpressionSyntax invocation, SemanticModel semanticModel, LanguageVersion languageVersion)
         {
-            if (!(semanticModel.GetSymbolInfo(invocation).Symbol is IMethodSymbol methodSymbol) ||
-                !methodSymbol.IsDisposeMethod())
+            if (semanticModel.GetSymbolInfo(invocation).Symbol is not IMethodSymbol methodSymbol
```

Comment:
This is an example of very bad method, that I have trouble to read. It basically says so much what it isn't, that I can't tell what it is. Luckily, we can fix that by making all conditions positive to say what it is directly. And then, it can be an arrow method:
```
        private static bool IsDisposeMethodCalled(InvocationExpressionSyntax invocation, SemanticModel semanticModel, LanguageVersion languageVersion) =>
            semanticModel.GetSymbolInfo(invocation).Symbol is IMethodSymbol methodSymbol
            && methodSymbol.IsDisposeMethod()
            && semanticModel.Compilation.GetTypeMethod(SpecialType.System_IDisposable, DisposeMethodName) is { }  disposeMethodSignature
            && (methodSymbol.Equals(methodSymbol.ContainingType.FindImplementationForInterfaceMember(disposeMethodSignature))
                || methodSymbol.ContainingType.IsDisposableRefStruct(languageVersion));
```